File: lib-android/src/main/java/com/pranav/lib_android/task/JavaBuilder.java
package com.pranav.lib_android.task;

import android.content.Context;

import com.pranav.lib_android.interfaces.*;
import com.pranav.lib_android.task.java.*;

public class JavaBuilder extends Builder {

    public final ClassLoader classloader;

    public Context mContext;

    public JavaBuilder(Context context, ClassLoader loader) {
        mContext = context;
        classloader = loader;
    }

    @Override
    public Context getContext() {
        return mContext;
    }

    @Override
    public ClassLoader getClassloader() {
        return this.classloader;
    }
}

File: lib-android/src/main/java/com/pranav/lib_android/task/JavaBuilder.kt
package com.pranav.lib_android.task

import android.content.Context
import com.pranav.lib_android.interfaces.Builder

class JavaBuilder constructor(
    val mContext: Context,
    val classLoader: ClassLoader
  ): Builder() {

	override fun getContext() = mContext

	override fun getClassloader() = classloader
}
